library/include/BitArray.h

```diff
@@ -526,27 +407,125 @@ namespace DFHack
 
         iterator begin()
         {
-            return iterator(static_cast<L *>(this), static_cast<L *>(this));
+            return iterator(static_cast<L *>(this), static_cast<L *>(this)->next);
         }
         const_iterator begin() const
         {
-            return const_iterator(static_cast<L *>(this), static_cast<L *>(this));
+            return const_iterator(static_cast<L *>(this), static_cast<L *>(this)->next);
         }
         const_iterator cbegin() const
         {
             return begin();
         }
         iterator end()
         {
-            return iterator(static_cast<L *>(this), nullptr, false);
+            return iterator(static_cast<L *>(this), nullptr);
         }
         const_iterator end() const
         {
-            return const_iterator(static_cast<L *>(this), nullptr, false);
+            return const_iterator(static_cast<L *>(this), nullptr);
         }
         const_iterator cend() const
         {
             return end();
         }
+
+        iterator erase(const_iterator pos)
+        {
+            auto root = static_cast<L *>(this);
+            CHECK_INVALID_ARGUMENT(pos.iter.root == root);
+            CHECK_NULL_POINTER(pos.iter.cur);
+
+            auto link = pos.iter.cur;
+            auto next = link->next;
+            if (link->next && link->prev)
+            {
+                link->next->prev = link->prev;
+                link->prev->next = link->next;
+            }
+            else if (link->next)
+            {
+                link->next->prev = nullptr;
+            }
+            else if (link->prev)
+            {
+                link->prev->next = nullptr;
+            }
+
+            if (root->next == link)
+            {
+                root->next = next;
+            }
+
```

Comment:
Can we reduce branches?

Maybe something like following would work:
<pre>
if (link->prev) link->prev->next = link->next;
else root->next = link->next;
if (link->next) link->next->prev = link->prev;
</pre>